pkg/api/environments.go

```diff
@@ -0,0 +1,146 @@
+package api
+
+import (
+	"github.com/google/uuid"
+	"github.com/permitio/permit-golang/pkg/config"
+	"github.com/permitio/permit-golang/pkg/errors"
+	"github.com/permitio/permit-golang/pkg/models"
+	"github.com/permitio/permit-golang/pkg/openapi"
+	"go.uber.org/zap"
+	"golang.org/x/net/context"
+)
+
+type Environments struct {
+	permitBaseApi
+}
+
+func NewEnvironmentsApi(client *openapi.APIClient, config *config.PermitConfig) *Environments {
+	return &Environments{
+		permitBaseApi{
+			client: client,
+			config: config,
+			logger: config.Logger,
+		},
+	}
+}
+
+// List the environments in the project of your context
+// Usage Example:
+// 	`environments, err := PermitClient.Api.Environments.List(ctx, 1, 10)`
+func (e *Environments) List(ctx context.Context, page int, perPage int) ([]models.EnvironmentRead, error) {
```

Comment:
Hmm... It would be really cool if we'll have an idea to eliminate that code duplication.

I agree that sometimes readability beats reusability - but I feel like we can win both here...

Maybe we can have all APIs (e.g. `EnvironmentsApi`, `UsersApi`) implement some "crud" interface (which has List, Create etc..) - we could wrap it once

Reply:
I agree in general.
As I showed you today, I want our users (developers) to be able to jump into the reference and understand immediately what the functions are expecting and returning, without the need to understand generics nor the SDK code's logic.
So I'll leave it like this for the meanwhile, if you have any other ideas, this can help us with all of our SDKs